frontend/src/metabase/dashboard/components/DashCard/DashCardMenu/DashCardMenu.tsx

```diff
@@ -170,7 +178,8 @@ DashCardMenu.shouldRender = ({
     !isPublic &&
     !isEditing &&
     !isXray &&
-    (canEditQuestion(question) || canDownloadResults(result))
+    (getHasDataAccess(question.metadata().databasesList()) ||
```

Comment:
This is not 100% the same but it should be correct in 99% real world cases. In the worst case users would get an empty menu.